app/src/main/java/com/pronoidsoftware/tasky/di/AppModule.kt

```diff
@@ -0,0 +1,28 @@
+package com.pronoidsoftware.tasky.di
+
+import android.content.SharedPreferences
+import androidx.security.crypto.EncryptedSharedPreferences
+import androidx.security.crypto.MasterKey
+import com.pronoidsoftware.tasky.TaskyApp
+import dagger.Module
+import dagger.Provides
+import dagger.hilt.InstallIn
+import dagger.hilt.components.SingletonComponent
+import javax.inject.Singleton
+
+@Module
+@InstallIn(SingletonComponent::class)
+object AppModule {
+
+    @Provides
+    @Singleton
+    fun provideSharedPreferences(): SharedPreferences {
+        return EncryptedSharedPreferences(
+            TaskyApp(),
+            "auth_pref",
+            MasterKey(TaskyApp()),
+            EncryptedSharedPreferences.PrefKeyEncryptionScheme.AES256_SIV,
+            EncryptedSharedPreferences.PrefValueEncryptionScheme.AES256_GCM,
+        )
+    }
```

Comment:
To answer your question regarding deprecation, I did hear about EncryptedSharedPreferences becoming deprecated (sigh 😞). It was always the easiest way to encrypt persistent storage.. There's no real replacement AFAIK, other than writing encryption logic manually for DataStore. Got a video about that:
https://youtu.be/D_q07P1sfcc?si=y_Dr1AXOvZ0qzy1y

If I remember correctly there's some sort of bug in that video's code which makes the implementation break with too long inputs, so make sure to check that..

Reply:
Awesome, thanks for the suggestion! I'll have to check this one out. I'll leave it as-is for now so that I can move on, but will try to apply it once the main functionality of the app is complete 👍